### PR TITLE
491 compacted expanded urls

### PR DESCRIPTION
**Pull request by julienmachon**
:boom: 
Screenshot from 2019-03-20 10-47-46
Screenshot from 2019-03-20 10-47-52

### CODE REVIEW

File: src/shared/components/Resources/MetadataCard.tsx [hunk 1]
```diff
@@ -32,7 +32,13 @@ const ResourceMetadataCard: React.FunctionComponent<
     type,
     rev,
   } = props;
-  const [userName] = createdBy.split('/').slice(-1);
+  let userName = createdBy;
+  // let's try to extract a nice looking user name
+  try {
+    [userName] = createdBy.split('/').slice(-1);
```
Comment (mdupont) [inline src/shared/components/Resources/MetadataCard.tsx]: Could use a test ;-)

File: src/shared/components/Resources/ResourceEditor.tsx [hunk 2]
```diff
@@ -68,7 +70,14 @@ const ResourceEditor: React.FunctionComponent<ResourceEditorProps> = props => {
           )}
         </div>
         <div className="controls">
-          {isEditing && valid && (
+          {!isEditing && valid && (
+            <Switch
+              checkedChildren="expanded"
+              unCheckedChildren="expand"
```
Comment (mdupont) [inline src/shared/components/Resources/ResourceEditor.tsx]: to test as well?

File: src/shared/components/User/Avatar.tsx [hunk 3]
```diff
@@ -7,7 +7,12 @@ export interface AvatarProps {
 
 const UserAvatar: React.FunctionComponent<AvatarProps> = props => {
   const { createdBy } = props;
-  const [name] = createdBy.split('/').slice(-1);
+  let name = createdBy;
+  try {
+    [name] = createdBy.split('/').slice(-1);
```
Comment (mdupont) [inline src/shared/components/User/Avatar.tsx]: same

File: src/shared/store/actions/nexus/resource.ts [hunk 4]
```diff
@@ -73,8 +74,15 @@ export const fetchAndAssignResource: ActionCreator<ThunkAction> = (
   > => {
     dispatch(fetchResourceAction());
     try {
-      const project: Project = await Project.get(orgLabel, projectLabel);
-      const resource = await project.getResource(resourceId);
+      const resource = await Resource.get(
+        orgLabel,
+        projectLabel,
+        '_',
```
Comment (mdupont) [inline src/shared/store/actions/nexus/resource.ts]: Implicit magic value, should be a constant or come from a config

Comment (julienmachon) [reply src/shared/store/actions/nexus/resource.ts]: fixed in e20dbed86139332f97391b9ce31b78f434665d2d

File: src/shared/components/User/Avatar.tsx (hunk 3 above)
Comment (julienmachon) [reply src/shared/components/User/Avatar.tsx]: added tests in 759c033fceb7fcf8ad141fea41c50e61ee195945

File: src/shared/components/Resources/ResourceEditor.tsx (hunk 2 above)
Comment (julienmachon) [reply src/shared/components/Resources/ResourceEditor.tsx]: added tests in 759c033fceb7fcf8ad141fea41c50e61ee195945

**Pull request merged**